apps/emqx_durable_storage/src/emqx_ds.erl

```diff
@@ -537,6 +580,12 @@ count(DB) ->
 timestamp_us() ->
     erlang:system_time(microsecond).
 
+%% @doc Version of `emqx_topic:words()' that doesn't transform empty
+%% topic into atom ''
+-spec topic_words(binary()) -> topic().
+topic_words(<<>>) -> [];
+topic_words(Bin) -> emqx_topic:words(Bin).
```

Comment:
Q: is the intention here really to prevent only `<<>>` from becoming `['']`?  Non-empty topics with empty levels will also violate the `emqx_ds:topic()` typespec by returning `''` atoms instead of always being `list(binary())`.

Ex:

```elixir
iex(emqx@127.0.0.1)11> :emqx_topic.words("/")
[:"", :""]
```

Reply:
Right... That's a copy of an old function from skipstream_lts, though. I wouldn't touch it just yet. Instead clarify the behavior everywhere.